### PR TITLE
fix: reinforce search/export of patron transaction 

### PR DESCRIPTION
**Pull request by zannkukai**
If patron transaction related resources are delete, the search and export caused error. This PR fix all these problems.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>
Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

### CODE REVIEW

File: rero_ils/modules/patron_transaction_events/serializers/csv.py [hunk 1]
```diff
@@ -114,6 +115,15 @@ def generate_csv():
         return stream_with_context(generate_csv())
 
 
+def _skip_if_no_record(func):
```
Comment (jma) [inline rero_ils/modules/patron_transaction_events/serializers/csv.py:118]: A method starting with '_' is considered as private. What does it means for a function?

Comment (jma) [reply rero_ils/modules/patron_transaction_events/serializers/csv.py:118]: This function can be generalized and added as a generic decorator.

**Review by lauren-d — APPROVED** [review]

**Review by jma — APPROVED** [review]
Please change your commit message as defined in the guide: https://github.com/rero/developer-resources/blob/e8be0aab1cdc1d2a5b50782807e38dda72a54ffd/git-github/git.md#subject

**Review by Garfield-fr — APPROVED** [review]

**Review by benerken — APPROVED** [review]

File: rero_ils/modules/patron_transaction_events/serializers/csv.py (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/patron_transaction_events/serializers/csv.py:118]: @jma due to the default return statement, I think this decorator isn't so generic and can't be generalized.

**Pull request merged**